crates/network-scanner/src/ping.rs

```diff
@@ -0,0 +1,54 @@
+use std::net::Ipv4Addr;
+
+use network_scanner_net::tokio_raw_socket::TokioRawSocketStream;
+use network_scanner_proto::icmp_v4;
+
+use crate::NetworkScanError;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+type NetowrkScanResult<T> = Result<T, NetworkScanError>;
+
+pub async fn ping(ip: impl Into<Ipv4Addr>) -> NetowrkScanResult<()> {
+    let mut socket = TokioRawSocketStream::connect(ip)
+        .await
+        .map_err(|e| NetworkScanError::IoError(e))?;
+
+    let time = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .map_err(|e| NetworkScanError::Other(format!("Failed to get time: {}", e)))?
```

Comment:
nitpick: [Error messages should start by a lower case letter](https://github.com/Devolutions/IronRDP/blob/33c66edfc6ddc8fd47d83b3ef4ddf9b1875a4bc2/STYLE.md#formatting-of-error-messages).

This applies to all the places, including the `.expect("…")`.

Reply:
updated